pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -370,8 +370,9 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 
 	service.Spec.Selector = controller.IngressControllerDeploymentPodSelector(ci).MatchLabels
 
-	lb := ci.Status.EndpointPublishingStrategy.LoadBalancer
-	isInternal := lb != nil && lb.Scope == operatorv1.InternalLoadBalancer
+	lbSpec := ci.Spec.EndpointPublishingStrategy.LoadBalancer
```

Comment:
double check for me, but I think this could be a NIL pointer exception if you don't have `spec.EndpointPublishingStrategy` set. I don't think `spec` gets "defaulted" anywhere unlike status.

Reply:
I've added a check.